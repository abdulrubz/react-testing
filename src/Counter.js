import React, { Component } from "react";

export default class Counter extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }

  makeIncrementer = (amount) => () => {
    this.setState((prevState) => {
      return {
        count: prevState.count + amount,
      };
    });
  };

  increment = this.makeIncrementer(1);

  decrement = this.makeIncrementer(-1);

  render() {
    return (
      <div>
        <h1>Count: {this.state.count}</h1>
        <button className="increment" onClick={this.increment}>
          ++
        </button>
        <button className="decrement" onClick={this.decrement}>
          --
        </button>
      </div>
    );
  }
}
