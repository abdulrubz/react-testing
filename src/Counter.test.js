import React from "react";
import Counter from "./Counter";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("Counter component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("starts with 0", () => {
    const wrapper = shallow(<Counter />);
    const text = wrapper.find("h1").text();
    expect(text).toEqual("Count: 0");
  });

  it("should increment count when Increment button is clicked", () => {
    const wrapper = shallow(<Counter />);
    const incrementBtn = wrapper.find("button.increment");
    incrementBtn.simulate("click");
    const text = wrapper.find("h1").text();
    expect(text).toEqual("Count: 1");
  });

  it("shou;d decrement count when Decrement button is clicked", () => {
    const wrapper = shallow(<Counter />);
    const decrementBtn = wrapper.find("button.decrement");
    decrementBtn.simulate("click");
    const text = wrapper.find("h1").text();
    expect(text).toEqual("Count: -1");
  });
});
