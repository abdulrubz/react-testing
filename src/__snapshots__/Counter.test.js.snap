// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches the snapshot 1`] = `
<div>
  <h1>
    Count: 
    0
  </h1>
  <button
    className="increment"
    onClick={[Function]}
  >
    ++
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    --
  </button>
</div>
`;
